File: backend/src/main/java/com/campus/learningspace/entity/Classroom.java
package com.campus.learningspace.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("classroom")
public class Classroom {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long buildingId;

    private String name;

    private String roomNumber;

    private Integer floor;

    private Integer type;

    private Integer capacity;

    private String equipment;

    private BigDecimal latitude;

    private BigDecimal longitude;

    private Integer checkinRadius;

    private Integer status;

    private Integer realTimeStatus;

    private BigDecimal environmentScore;

    private Integer totalReviews;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/classroom.ts
import request from '@/utils/request'

export interface Classroom {
  id?: number
  buildingId?: number
  name: string
  roomNumber?: string
  floor?: number
  type?: number
  capacity?: number
  equipment?: string
  latitude?: number
  longitude?: number
  checkinRadius?: number
  status?: number
  realTimeStatus?: number
  environmentScore?: number
  totalReviews?: number
  createTime?: string
  updateTime?: string
  deleted?: number
}

export interface Result<T> {
  code: number
  message: string
  data: T
}

/** 教室详情（含热度星级、综合评分均值、签到次数） */
export interface ClassroomDetailVO {
  classroom: Classroom
  /** 热度星级 1-5 */
  popularityStars: number
  /** 综合评分（历史评分平均值） */
  averageScore: number | null
  /** 历史评价条数 */
  totalReviews: number
  /** 签到次数 */
  checkinCount: number
}

export const getClassroomList = () => {
  return request.get<any, Result<Classroom[]>>('/classroom/list')
}

export const getClassroomsByBuilding = (buildingId: number) => {
  return request.get<any, Result<Classroom[]>>(`/classroom/building/${buildingId}`)
}

export const getAvailableClassrooms = (type?: number) => {
  return request.get<any, Result<Classroom[]>>('/classroom/available', { params: { type } })
}

export const getClassroomById = (id: number) => {
  return request.get<any, Result<Classroom>>(`/classroom/${id}`)
}

export const getClassroomDetail = (id: number) => {
  return request.get<any, Result<ClassroomDetailVO>>(`/classroom/${id}/detail`)
}

export const saveClassroom = (classroom: Classroom) => {
  return request.post<any, Result<boolean>>('/classroom', classroom)
}

export const updateClassroom = (classroom: Classroom) => {
  return request.put<any, Result<boolean>>('/classroom', classroom)
}

export const deleteClassroom = (id: number) => {
  return request.delete<any, Result<boolean>>(`/classroom/${id}`)
}
